plugins/snapshot/src/main.ts

```diff
@@ -288,6 +288,15 @@ class SnapshotMixin extends SettingsMixinDeviceBase<Camera> implements Camera {
             type: 'source',
             event: options?.reason === 'event',
         }, eventSnapshot ? 0 : 4000, async () => {
+            // If battery cam and not online, skip otherwise will wake up'
+            const realDevice = systemManager.getDeviceById<Online>(this.id);
+            if (this.mixinDeviceInterfaces.includes(ScryptedInterface.Battery) && !realDevice.online) {
+                return {
```

Comment:
I don't think this is valid return value. it should `throw new Error('not taking snapshot for battery camera')`